documentation/docs/client/moderation_tools.md

```diff
@@ -0,0 +1,195 @@
+    } 
+} 
+ 
+client.unmuteUser("user-id").enqueue { result -> 
+    if (result.isSuccess) { 
+        // User was unmuted 
+    } else { 
+        // Handle result.error() 
+    } 
+}
+```
+
+After muting a user messages will still be delivered via web-socket. Implementing business logic such as hiding messages from muted users or display them differently is left to the developer to implement.
+
+> Messages from muted users are not delivered via push (APN/Firebase)
+
+## Banning a User
+
+Users can be banned from an app entirely or from a channel. When a user is banned, it will not be allowed to post messages until the ban is removed or expired but it will be able to connect to Chat and to channels as before.
+
+In most cases, only admins or moderators are allowed to ban other users from a channel.
+
+| Name | Type | Description | Default | Optional |
+| :--- | :--- | :--- | :--- | :--- |
+| timeout | Int | The timeout in minutes until the ban is automatically expired. | 	no limit | &check; |
+| reason | String | The reason that the ban was created. | | &check; |
+
+> Banning a user from all channels can only be done using server-side auth.
+
+```kotlin
+// Ban user for 60 minutes from a channel 
+channelClient.banUser(targetId = "user-id", reason = "Bad words", timeout = 60).enqueue { result -> 
+    if (result.isSuccess) { 
+        // User was banned 
+    } else { 
+        // Handle result.error() 
+    } 
+} 
+ 
+channelClient.unBanUser(targetId = "user-id").enqueue { result -> 
+    if (result.isSuccess) { 
+        // User was unbanned 
+    } else { 
+        // Handle result.error() 
+    } 
+}
+```
+
+## Shadow Banning a User
+
+Users can be shadow banned from an app entirely or from a channel. When a user is shadow banned, it will still be allowed to post messages, but any message sent during, will have `shadowed: true` field. However, this will be invisible to the author of the message.
+
+> It's up to the client-side implementation to hide or otherwise handle these messages appropriately.
+
+```kotlin
+// Shadow ban user for 60 minutes from a channel 
+channelClient.shadowBanUser(targetId = "user-id", reason = "Bad words", timeout = 60).enqueue { result -> 
+     if (result.isSuccess) { 
+         // User was shadow banned 
+     } else { 
+         // Handle result.error() 
+     } 
+} 
+ 
+channelClient.removeShadowBan("user-id").enqueue { result -> 
+    if (result.isSuccess) { 
+        // Shadow ban was removed 
+    } else { 
+        // Handle result.error() 
+    } 
+}
+```
+
+Administrators can view shadow banned user status in `queryChannels()`, `queryMembers()` and `queryUsers()`.
+
+## Retrieving Banned Users
+
+Banned users can be retrieved in different ways:
+1. Using the dedicated query bans endpoint
+2. User Search: you can add the `banned:true` condition to your search. Please note that this will only return users that were banned at the app-level and not the ones that were banned only on channels.
```

Comment:
```suggestion
2. User Search: you can add the `banned:true` condition to your search. Please note that this will only return users that were banned at the app-level and not the ones that were banned only in channels.
```